xmtp_mls/src/groups/tests/test_remote_commit_log.rs

```diff
@@ -936,4 +959,368 @@ async fn test_bad_signature_handling() {
+
+    // Bo and Charlie sync to get the metadata update
+    bo_group.sync().await.unwrap();
+    charlie_group.sync().await.unwrap();
+
+    println!("✓ Bo and Charlie synced to get metadata updates");
+
+    // Verify bo and charlie now have the signing key in their mutable metadata
+    let bo_metadata = bo_group.mutable_metadata().unwrap();
+    let bo_signer = bo_metadata.commit_log_signer();
+
+    let charlie_metadata = charlie_group.mutable_metadata().unwrap();
+    let charlie_signer = charlie_metadata.commit_log_signer();
+
+    assert!(
+        bo_signer.is_some(),
+        "Bo should have signing key in mutable metadata after sync"
+    );
+    assert!(
+        charlie_signer.is_some(),
+        "Charlie should have signing key in mutable metadata after sync"
+    );
+
+    // Verify all parties have the same new signing key
+    assert_eq!(
+        bo_signer.unwrap().as_slice(),
+        new_signing_key.as_slice()
+    );
+    assert_eq!(
+        charlie_signer.unwrap().as_slice(),
+        new_signing_key.as_slice()
+    );
+
+    println!("✓ All participants now have the new signing key in mutable metadata");
+
+    // Additional verification: the consensus key should be set in the database
+    let stored_group = alix
+        .context
+        .db()
+        .find_group(&group.group_id)?
+        .unwrap();
+    assert_eq!(stored_group.commit_log_public_key, Some(new_public_key.clone()));
+
+    println!("✓ Consensus public key correctly stored in database");
+
+    // Final verification: all parties should be able to verify signatures with this key
+    let test_entry = xmtp_proto::xmtp::mls::message_contents::CommitLogEntry {
+        sequence_id: 999,
+        serialized_commit_log_entry: b"test message".to_vec(),
+        signature: None,
+    };
+
+    let signature_bytes = crypto
+        .sign(
+            openmls::prelude::SignatureScheme::ED25519,
+            &test_entry.serialized_commit_log_entry,
+            new_signing_key.as_slice(),
+        )
+        .unwrap();
+
+    let signed_entry = xmtp_proto::xmtp::mls::message_contents::CommitLogEntry {
+        sequence_id: 999,
+        serialized_commit_log_entry: test_entry.serialized_commit_log_entry.clone(),
+        signature: Some(
+            xmtp_proto::xmtp::identity::associations::RecoverableEd25519Signature {
+                public_key: new_public_key.clone(),
+                bytes: signature_bytes,
+            },
+        ),
+    };
+
+    // All parties should be able to verify this signature
+    assert!(crypto
+        .verify_commit_log_signature(&signed_entry, &new_public_key)
+        .is_ok());
+
+    println!("✓ All parties can verify signatures with the discovered signing key");
+    println!("Test passed: Legacy group signing key discovery via remote commit log works correctly");
+}
+
 // TODO(rich): E2E test for signing key creation and verification on legacy groups
```

Comment:
think we can remove this todo since we have `test_legacy_group_signing_key_discovery_via_remote_commit_log` unless there was another test you had in mind?